lib/fs.js

```diff
@@ -465,11 +465,16 @@ fs.readFileSync = function(path, options) {
 
 // Used by binding.open and friends
 function stringToFlags(flag) {
-  // Only mess with strings
-  if (typeof flag !== 'string') {
+  // Integers are already good to go
+  if (Number.isInteger(flag)) {
     return flag;
   }
 
+  // We only allow strings from here on
+  if (typeof flag !== 'string') {
+    throw new TypeError('File open flags may only be string or integer');
```

Comment:
Anyone who sees this message and looks at the docs would immediately report think this a bug, since it conflicts with the docs. If you want to say that, I suggest documenting the behaviour. 

I strongly suspect that the fact that flags can be an int is historical, kept for backwards compat from the v0.small days, and should be deprecated, but its hard to deprecate a piece of an API.